plugins/mongo_db/mongo_db_operations.cpp

```diff
@@ -0,0 +1,1249 @@
+#include <golos/plugins/mongo_db/mongo_db_operations.hpp>
+#include <golos/plugins/follow/follow_objects.hpp>
+#include <golos/plugins/follow/plugin.hpp>
+#include <golos/plugins/chain/plugin.hpp>
+#include <golos/chain/comment_object.hpp>
+#include <golos/chain/account_object.hpp>
+
+#include <bsoncxx/builder/stream/array.hpp>
+#include <bsoncxx/builder/stream/value_context.hpp>
+#include <bsoncxx/builder/basic/document.hpp>
+#include <appbase/plugin.hpp>
+
+
+namespace golos {
+namespace plugins {
+namespace mongo_db {
+
+    using bsoncxx::builder::stream::array;
+    using bsoncxx::builder::stream::document;
+    using namespace golos::plugins::follow;
+
+    // Helper functions
+    document format_authority(const authority& auth) {
+        array account_auths_arr;
+        for (const auto& iter : auth.account_auths) {
+            document temp;
+            temp << "public_key_type" << (std::string)iter.first
+                 << "weight_type" << std::to_string(iter.second);
+            account_auths_arr << temp;
+        }
+        array key_auths_arr;
+        for (auto& iter : auth.key_auths) {
+            document temp;
+            temp << "public_key_type" << (std::string)iter.first
+                 << "weight_type" << std::to_string(iter.second);
+            key_auths_arr << temp;
+        }
+
+        document authority_doc;
+        authority_doc << "owner" << auth.owner
+                      << "weight_threshold" << std::to_string(auth.weight_threshold);
+        authority_doc << "account_auths" << account_auths_arr;
+        authority_doc << "key_auths" << key_auths_arr;
+
+        return authority_doc;
+    }
+
+    document format_chain_properties(const chain_properties& props) {
+        document props_doc;
+        props_doc << "account_creation_fee" << props.account_creation_fee.to_string()
+                  << "maximum_block_size" << std::to_string(props.maximum_block_size)
+                  << "sbd_interest_rate" << std::to_string(props.sbd_interest_rate);
+        return props_doc;
+    }
+
+    void format_asset(const asset& _asset, document& asset_doc, const std::string& val) {
+
+        asset_doc << val + "_value" << std::to_string(_asset.to_real());
+        asset_doc << val + "_symbol" << _asset.symbol_name();
+    }
+
+    void format_value(document& doc, const std::string& name, const std::string& value) {
```

Comment:
Should allow to store integers/doubles/...  Not only strings. It is impossible to aggregate data in database if all fields is strings.